### PR TITLE
feat: web3 tab for facebook

### PR DESCRIPTION
**Pull request by yanzhihong23**
## Description

Closes  [MASK-750](https://mask.atlassian.net/browse/MASK-750)

## Type of change

- [ ] Documentation
- [ ] Code refactoring (Restructuring existing code w/o changing its observable behavior)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (a fix or feature that would make something no longer possible to do/require old user must upgrade their Mask Network to this new version)

## Checklist

- [ ] My code follows the style guidelines of this project.
- [ ] I have performed a self-review of my own code.
  - [ ] I have removed all in development `console.log`s
  - [ ] I have removed all commented code.
- [ ] I have commented on my code, particularly in hard-to-understand areas.
- [ ] I have read [Internationalization Guide](https://github.com/DimensionDev/Maskbook/blob/develop/docs/i18n-guide.md) and moved text fields to the i18n JSON file.

If this PR depends on external APIs:

- [ ] I have configured those APIs with CORS headers to let extension requests get passed. 
  - chrome extension: `chrome-extension://[id]`
  - firefox extension: `moz-extension://[id]`
- [ ] I have delegated all web requests to the background service via the internal RPC bridge.

### CODE REVIEW

File: packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts [hunk 1]
```diff
@@ -80,3 +80,32 @@ export const searchFacebookSaveAvatarButtonSelector = () =>
...
+export const profileTabSelectedSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[data-pagelet="ProfileTabs"] [aria-selected="true"]')
+
+// fot inserting web3 tab
+export const searchProfileTabSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[data-pagelet="ProfileTabs"] a:nth-child(7)')
+
+// fot getting the inserted web3 tab
+export const web3TabSelector: () => LiveSelector<E, true> = () =>
+    querySelector<E>('[data-pagelet="ProfileTabs"] a:nth-child(7)+span')
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/selector.ts]:
```suggestion
    querySelector<HTMLSpanElement>('[data-pagelet="ProfileTabs"] a:nth-child(7)+span')
```

File: packages/mask/src/social-network-adaptor/facebook.com/injection/ProfileTab.tsx [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+    )
+}
+
+export function injectProfileTabAtFacebook(signal: AbortSignal) {
+    const watcher = new MutationObserverWatcher(searchProfileTabSelector())
+    startWatch(watcher, signal)
+    createReactRootShadowed(watcher.firstDOMProxy.afterShadow, { signal }).render(<ProfileTabAtFacebook />)
+
+    setTimeout(() => {
+        const web3Tab = web3TabSelector().evaluate() as HTMLSpanElement
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/injection/ProfileTab.tsx]:
```suggestion
        const web3Tab = web3TabSelector().evaluate()
```

File: packages/mask/src/social-network-adaptor/facebook.com/injection/ProfileContent.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    startWatch(watcher, signal)
+    createReactRootShadowed(watcher.firstDOMProxy.beforeShadow, { signal }).render(<ProfileTabContentAtFacebook />)
+}
+
+export function injectProfileTabContentAtFacebook(signal: AbortSignal) {
+    injectProfileTabContentState(signal)
+}
+
+function getStyleProps() {
+    const EMPTY_STYLE = {} as CSSStyleDeclaration
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/injection/ProfileContent.tsx:18]:
```suggestion
    const EMPTY_STYLE: CSSStyleDeclaration = {}
```

**Review by guanbinrui — DISMISSED** [review]

**Comment by wenluomask** [comment]
Checked.

File: packages/mask/src/social-network-adaptor/facebook.com/utils/user.ts [hunk 4]
```diff
@@ -66,3 +70,19 @@ export const getAvatarId = (avatarURL: string) => {
...
     return match[1]
 }
+
+export const getPersonalHomepage = () => {
+    const intro = searchIntroSectionSelector().evaluate()
+    const node = searchHomepageSelector().evaluate()
+    if (intro && node) {
+        let text = collectNodeText(node)
+        if (text && !text.startsWith('http')) {
+            text = 'http://' + text
```
Comment (guanbinrui) [inline packages/mask/src/social-network-adaptor/facebook.com/utils/user.ts:80]: Why not HTTPS?

**Pull request merged**